### ISSUE TITLE
linuxbrew is not working properly under CentOS6.5

### ISSUE THREAD

**Issue by xq262144** [Homebrew/linuxbrew#689]
I tried `brew install curl --with-c-ares` in my CentOS 6.5 box and got unexpected errors.
```
==> Installing dependencies for curl: pkg-config, makedepend, zlib, openssl
==> Installing curl dependency: pkg-config
Error: pkg-config cannot be built with any available compilers.
To install this formula, you may need to:
  brew install gcc
```

However, `gcc 4.4.7` is shipped with the CentOS 6.5 system and working fine.
```
Using built-in specs.
Target: x86_64-redhat-linux
Configured with: ../configure --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-bootstrap --enable-shared --enable-threads=posix --enable-checking=release --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-languages=c,c++,objc,obj-c++,java,fortran,ada --enable-java-awt=gtk --disable-dssi --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-1.5.0.0/jre --enable-libgcj-multifile --enable-java-maintainer-mode --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --disable-libjava-multilib --with-ppl --with-cloog --with-tune=generic --with-arch_32=i686 --build=x86_64-redhat-linux
Thread model: posix
gcc version 4.4.7 20120313 (Red Hat 4.4.7-11) (GCC)
```
Calling `gcc --version`
```
gcc (GCC) 4.4.7 20120313 (Red Hat 4.4.7-11)
Copyright (C) 2010 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

And I found there is a failure gcc version regex search in
https://github.com/Homebrew/linuxbrew/blob/master/Library/Homebrew/os/mac.rb#L153

I simply rewrite the regex from
```ruby
/gcc(?:-\d(?:\.\d)? \(.+\))? (\d\.\d\.\d)/
```
to
```ruby
/gcc(?:-\d\.\d)? (?:\(.+\))? (\d\.\d\.\d)/
```
and it works.

But I'm not pretty sure if that works for other Linux distro.

So ...

**Comment by jervisfm** [comment]
A possible workaround for that is to add symlinks to the gcc compiler under the bin/ sub-directory of linuxbrew . From https://github.com/Homebrew/linuxbrew/wiki/Standalone-Installation#to-use-gcc-44-on-centos-5

```
sudo yum install gcc44 gcc44-c++
ln -s /usr/bin/gcc44 `brew --prefix`/bin/gcc-4.4
ln -s /usr/bin/g++44 `brew --prefix`/bin/g++-4.4
ln -s /usr/bin/gfortran44 `brew --prefix`/bin/gfortran-4.4
````

**Comment by sjackman** [comment]
This issue should be resolved by e61395214440500656a208c2595d37bc62c9aa94
Thanks for the bug report, Qi.
